pkg/controller/servicemesh/webhooks/validation/controlplane_upgrade_v1_0.go

```diff
@@ -0,0 +1,117 @@
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "apikey"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "authorization"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "checknothing"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "kubernetes"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "listentry"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "logentry"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "edge"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "metric"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "quota"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "reportnothing"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "tracespan"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "cloudwatch"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "dogstatsd"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "zipkin"},
+	}
+)
+
+func (v *ControlPlaneValidator) validateUpgradeFromV1_0(smcp *maistrav1.ServiceMeshControlPlane) error {
+	var allErrors []error
+
+	meshNamespaces := sets.NewString(smcp.GetNamespace())
+	smmr, err := v.getSMMR(smcp)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return pkgerrors.Wrap(err, "error retrieving ServiceMeshMemberRoll for mesh")
+		}
+	}
+	meshNamespaces.Insert(smmr.Status.ConfiguredMembers...)
+
+	// return error if any deprecated mixer resources are being used
+	for _, gvk := range unsupportedOldResourcesV1_1 {
+		objects := &unstructured.UnstructuredList{}
+		objects.SetGroupVersionKind(gvk)
+		// XXX: do we list all in the cluster, or list for each member namespace?
+		if err := v.client.List(context.TODO(), nil, objects); err != nil {
+			if !meta.IsNoMatchError(err) && !errors.IsNotFound(err) {
+				return pkgerrors.Wrapf(err, "error listing %s resources", gvk.String())
+			}
+		}
+		objects.EachListItem(func(obj runtime.Object) error {
+			metaObj, err := meta.Accessor(obj)
+			if err != nil {
+				allErrors = append(allErrors, pkgerrors.Wrapf(err, "error accessing object metadata for %s resource", gvk.String()))
+			}
+			// we only care about resources in this mesh, which aren't being managed by the operator directly
+			if meshNamespaces.Has(metaObj.GetNamespace()) && !metav1.IsControlledBy(metaObj, smcp) {
+				allErrors = append(allErrors, fmt.Errorf("%s/%s of type %s is not supported in newer version", metaObj.GetNamespace(), metaObj.GetName(), gvk.String()))
+			}
+			return nil
+		})
+	}
+
+	// Any service ports using 443 are using http/http2 in their name (http not allowed on port 443)
```

Comment:
Is this a meaningful check? This is more of a general configuration issue than something that comes up during upgrade, right?

Reply:
I guess it depends on your general philosophy of what we should be checking.  If the user is using http/http2 protocols on port 443, an upgrade will break their application.  The logic in these upgrade/downgrade checks is intended to ensure an upgrade or downgrade won't break their applications.  I think it would be beneficial for the operator to migrate things like this when possible (nothing in 1.0 <-> 1.1, but I could see migrating Policy to AuthorizationPolicy in the future).

Reply:
I just realized that we do check for `AuthorizationPolicy` on downgrade.  Should we add logic to migrate that to Policy/RbacConfig on downgrade?

Reply:
> I guess it depends on your general philosophy of what we should be checking. If the user is using http/http2 protocols on port 443, an upgrade will break their application. 

Didn't know this was the case! Makes total sense to check this then, sorry